x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/Operator.java

```diff
@@ -105,5 +105,21 @@ interface OperatorFactory extends Describable {
     /**
      * Status of an {@link Operator} to be returned by the tasks API.
      */
-    interface Status extends ToXContentObject, VersionedNamedWriteable {}
+    interface Status extends ToXContentObject, VersionedNamedWriteable {
+        /**
+         * The number of documents found by this operator. Most operators
+         * don't find documents and will return {@code 0} here.
+         */
+        default long documentsFound() {
+            return 0;
+        }
+
+        /**
+         * The number of values loaded by this operator. Most operators
+         * don't load values and will return {@code 0} here.
+         */
+        default long valuesLoaded() {
+            return 0;
+        }
+    }
```

Comment:
Not related to this change, but I think it is worth revisiting operator status.
Currently this is interface with implementation per each operator. It is used to expose some diagnostics attributes outside.

I believe we should replace it with a simple map (or maybe a wrapper on top of the map).
This way we would not need to have a custom implementation per operator, could have common equals/hashCode and toString logic. With map structure we will not need a new transport version when adding/renaming/removing infos.